### PR TITLE
Changing to use go mod

### PR DESCRIPTION
**Pull request by mtdavee**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,6 @@ vendor
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+# Build artifact, specifically for this project running go build
+alive-ssai-tracking-events-golang
```
Comment (tedk) [inline .gitignore]: The [Makefile](https://github.com/zencoder/alive-ssai-tracking-events-golang/blob/master/Makefile#L10) should be writing this to the `bin/` dir.

Comment (mtdavee) [reply .gitignore]: Yes, but when I was checking builds after setting up go mod, I was verifying with `go build ./...` which outputs to the base directory, so i figured it doesn't hurt to ignore it so no one commits on accident.

File: Makefile [hunk 2]
```diff
@@ -1,10 +1,56 @@
...
+cover: test $(COVER_LIST)
+$(COVER_LIST): %_cover:
+	go tool cover -html=$(COVERAGEDIR)/$(subst /,_,$*).coverprofile -o $(COVERAGEDIR)/$(subst /,_,$*).html
+
+.PHONY: assert-no-diff
+assert-no-diff:
+	test -z "$(shell git status --porcelain)"
+
+.PHONY: run
+ru n: build
```
Comment (tedk) [inline Makefile]: ` `

Comment (mtdavee) [reply Makefile]: dammit.

**Pull request merged**